Example/CustomPermissionsTableViewController.swift

```diff
@@ -0,0 +1,107 @@
+
+    var customPermissionHandlerObj: CustomPermissionsCompletionHandler?
+    var permissionsRequestList: [MASDKCustomPermissionModel]?
+    var miniAppTitle: String = "MiniApp"
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.tableView.tableFooterView = UIView(frame: .zero)
+        self.addBarButtonItems()
+        self.addFooterInfo()
+    }
+
+    func addBarButtonItems() {
+        let allowBarButtonItem = UIBarButtonItem(title: "Allow", style: .done, target: self, action: #selector(allowPermissions))
+        let dontAllowBarButtonItem = UIBarButtonItem(title: "Don't Allow", style: .done, target: self, action: #selector(dontAllowPermissions))
+        self.navigationItem.rightBarButtonItem  = allowBarButtonItem
+        self.navigationItem.leftBarButtonItem  = dontAllowBarButtonItem
+    }
+
+    @objc func allowPermissions() {
+        guard let permissionRequest = permissionsRequestList else {
+            return
+        }
+        customPermissionHandlerObj?(.success(permissionRequest))
+        self.dismiss(animated: true, completion: nil)
+    }
+
+    @objc func dontAllowPermissions() {
+        let alert = UIAlertController(title: "Are you sure you don't want to allow any of the permissions?", message: "", preferredStyle: .alert)
+        let yesAction = UIAlertAction(title: "Yes", style: .default) { (_) in
+            self.permissionsRequestList?.forEach {
+                $0.isPermissionGranted = .denied
+            }
+            self.customPermissionHandlerObj?(.success(self.permissionsRequestList!))
+            self.dismiss(animated: true, completion: nil)
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alert.addAction(cancelAction)
+        alert.addAction(yesAction)
+        self.present(alert, animated: true, completion: nil)
+    }
+
+    @objc func permissionValueChanged(_ sender: UISwitch) {
+        if permissionsRequestList?.indices.contains(sender.tag) ?? false {
+            let permissionModel = permissionsRequestList?[sender.tag]
+            if sender.isOn {
+                permissionModel?.isPermissionGranted = .allowed
+            } else {
+                permissionModel?.isPermissionGranted = .denied
+            }
+        }
+        let allPermissionsDenied = permissionsRequestList?.allSatisfy {
+            $0.isPermissionGranted == MiniAppCustomPermissionGrantedStatus.denied
+            } ?? false
+        if allPermissionsDenied {
+            self.navigationItem.rightBarButtonItem?.isEnabled = false
+        } else {
+            self.navigationItem.rightBarButtonItem?.isEnabled = true
+        }
+    }
+
+    func getSwitchView(tagValue: Int) -> UISwitch {
+        let switchView = UISwitch(frame: .zero)
+        switchView.setOn(true, animated: true)
+        switchView.tag = tagValue
+        switchView.addTarget(self, action: #selector(permissionValueChanged(_:)), for: .valueChanged)
+        return switchView
+    }
```

Comment:
I thought it would be great to have native behavior(like in iOS Settings cell-associated toggle buttons) where the cell selection is disabled and only the toggle is enabled. Adding the toggle in the storyboard, well I thought of using the default cells but I will definitely think about adding a custom cells 👍

Reply:
This is just a demo app, so you don't have to break all the good work just because of this advice. In this case the behavior you are attempting to achieve is maybe better with a XIB containing the permission view and a `UIStackView` inside a `UIScrollView` to deal with permissions display. But honestly I would also have gone with the UITableView 👍